jenkins/_jenkins.py

```diff
@@ -192,27 +192,42 @@ def child_of(file_path, url_path):
     return result
 
 
-def print_common_failure_reasons(build_data):
-    individual_failures = build_data[build_data['result'] == FAILURE]
+def classify(url):
+    """
+    Classify the failure of a url.
 
-    classifications = []
-    for url in individual_failures['url']:
-        path = child_of(BASE_DIR.child('logs'), url).child('testReport')
+    Considers all the information about the url, such as testReport
+    consoleText and tries to provide the best classification.
+
+    :param str url: a url of a build.
+    :return str: the classification.
+    """
+    path = child_of(BASE_DIR.child('logs'), url).child('testReport')
+    if path.exists():
+        return "Failed Test"
+    else:
+        path = child_of(BASE_DIR.child('logs'), url).child('consoleText')
         if path.exists():
-            classifications.append("Failed Test")
+            with path.open() as f:
+                return classify_build_log(f.read(), path)
         else:
-            path = child_of(BASE_DIR.child('logs'), url).child('consoleText')
-            if path.exists():
-                with path.open() as f:
-                    classifications.append(classify_build_log(f.read(), path))
-            else:
-                classifications.append("Missing log")
-
-    individual_failures['classification'] = pandas.Series(classifications, index=individual_failures.index)
+            return "Missing log"
+
+
+def print_common_failure_reasons(build_data):
+    individual_failures = build_data[build_data['result'] == FAILURE]
+
+    classifications = map(classify, individual_failures['url'])
```

Comment:
I *think* you'd be better off doing this:

```python
individual_failures['url'].map(classify)
```

As this lets pandas parallelize internally. In fact, maybe if you do this, you won't need to create a `Series` manually, _maybe_ even assigning would just work.